rllib/algorithms/ppo/tf/ppo_tf_rl_module.py

```diff
@@ -0,0 +1,223 @@
+from dataclasses import dataclass
+import gymnasium as gym
+from typing import Mapping, Any, List
+from ray.rllib.core.rl_module.rl_module import RLModuleConfig
+from ray.rllib.core.rl_module.tf.tf_rl_module import TfRLModule
+
+from ray.rllib.core.rl_module.encoder_tf import FCTfConfig
+
+from ray.rllib.utils.framework import try_import_tf, try_import_tfp
+
+from ray.rllib.utils.annotations import override
+
+from ray.rllib.utils.nested_dict import NestedDict
+
+from ray.rllib.models.specs.specs_dict import SpecDict
+from ray.rllib.models.specs.specs_tf import TFTensorSpecs
+from ray.rllib.policy.sample_batch import SampleBatch
+
+tf1, tf, tfv = try_import_tf()
+tfp = try_import_tfp()
+
+tf1.enable_eager_execution()
+
+
+@dataclass
+class PPOTfModuleConfig(RLModuleConfig):
+    """Configuration for the PPO module.
+
+    Attributes:
+        pi_config: The configuration for the policy network.
+        vf_config: The configuration for the value network.
+    """
+
+    pi_config: FCTfConfig = None
+    vf_config: FCTfConfig = None
+    observation_space: gym.Space = None
+    action_space: gym.Space = None
+
+
+class PPOTfModule(TfRLModule):
+    def __init__(self, config: PPOTfModuleConfig):
+        super().__init__()
+        self.policy = config.pi_config.build()
+        self.value_function = config.vf_config.build()
+        self._obs_space = config.observation_space
+        self._action_space = config.action_space
+        self._is_discrete = isinstance(self._action_space, gym.spaces.Discrete)
+        self._action_dim = (
+            self._action_space.shape[0]
+            if not self._is_discrete
+            else self._action_space.n
+        )
+
+    @override(TfRLModule)
+    def input_specs_train(self) -> SpecDict:
```

Comment:
Lets' not make the same mistake twice. We decided that the specs at RLModule level should just be checking the existence of the keys. So let's dogfood that. i.e. instead of returning the spec_dict object, return the list of keys that the input/output should contain. If succesfull we have to migrate ppo torch module to the same thing.